common/djangoapps/track/middleware.py

```diff
@@ -140,12 +140,11 @@ def enter_request_context(self, request):
             'ip': self.get_request_ip_address(request),
         }
         for header_name, context_key in META_KEY_TO_CONTEXT_KEY.iteritems():
-            context[context_key] = request.META.get(header_name, '')
+            # HTTP headers might can contain the information that include latin1 characters
+            # decoding this using latin1 scheme will prevent to raise UnicodeDecodeError when using
+            # json.dumps for tracking purpose.
```

Comment:
Some phrases don't make sense here.  I suggest:
   "HTTP headers may contain Latin1 characters.  Decoding using Latin1 encoding here 
avoids encountering UnicodeDecodeError exceptions when these header strings are output to tracking logs."